### PR TITLE
[Addon] add the azure-keyvault-csi addon

### PR DESCRIPTION
**Pull request by Kolossi**
### Description of your changes

Add an azure-keyvault-csi trait addon.

As discussed with @FogDong & @wonderflow .

### How has this code been tested?

Used in our own AKS cluster to add secrets from an azure KeyVault which can then be successfully viewed with:

```
kubectl exec -n vela-test myapp-pod-name -- cat /mnt/secrets-store/my-secret
```

### Checklist

I have:

- [x] Title of the PR starts with type (e.g. `[Addon]` , `[example]` or `[Doc]`).
- [ ] Updated/Added any relevant [documentation](https://kubevela.io/docs/reference/addons/overview) and [examples](../examples). 
    - `not updated docs as they are in main repo and this PR to experimental addon not yet accepted.  Example in README`
- [x] New addon should be put in [experimental](../experimental/addons).
- [ ] Update addon should modify the `version` in `metadata.yaml` to generate a new version. 
    - `no update, new addon, v0.0.1`
- [ ] Any changes about [verified](../addons) addons should be tested with CI [script](../test/e2e-test/hack/addon-vela-test.sh).
    - `experimental addon only`

### CODE REVIEW

File: experimental/addons/azure-keyvault-csi/metadata.yaml
Comment (wangyikewxgm) [inline experimental/addons/azure-keyvault-csi/metadata.yaml:4]: Do you mean kubernetes version requirement?

Comment (wonderflow) [inline experimental/addons/azure-keyvault-csi/metadata.yaml:4]:
```suggestion
  kubernetes: ">=v1.14.0"
```

**Review by wonderflow — APPROVED** [review]
great work, thanks

File: experimental/addons/azure-keyvault-csi/README.md
Comment (FogDong) [inline experimental/addons/azure-keyvault-csi/README.md:9]: Can we add the helm chart in the addon so that we can install this when we enable the addon?
You can use helm in addon like: https://github.com/kubevela/catalog/blob/master/addons/model-training/template.yaml#L9

File: experimental/addons/azure-keyvault-csi/metadata.yaml
Comment (Kolossi) [reply experimental/addons/azure-keyvault-csi/metadata.yaml:4]: @wonderflow @wangyikewxgm I did mean vela version, as I am writing this now using vela 1.14 and I havent tested it with any earlier version not do I plan to?

I don't actually know the k8s version requirement, though we are on 1.23, so leaving that out.

Does that make sense?

File: experimental/addons/azure-keyvault-csi/README.md
Comment (Kolossi) [reply experimental/addons/azure-keyvault-csi/README.md:9]: @FogDong I understand some logic in adding the helm install, but my concern is that there would need to be pass-through of all possible helm configuration values (see https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/charts/csi-secrets-store-provider-azure/README.md#configuration).

In addition, those with permission to install vela/addons may not have permission for the kube-system namespace on the cluster.

So the helm install during enabling the addon would need to be optional as well as fully configurable.

This is why it seems best to leave secrets store csi driver/azure keyvault provider as a separate action.

What do you think?

File: experimental/addons/azure-keyvault-csi/metadata.yaml
Comment (Kolossi) [reply experimental/addons/azure-keyvault-csi/metadata.yaml:4]: ah, it should be 1.4 not 1.14!

Comment (Kolossi) [reply experimental/addons/azure-keyvault-csi/metadata.yaml]: corrected in latest commit

File: experimental/addons/azure-keyvault-csi/README.md
Comment (FogDong) [reply experimental/addons/azure-keyvault-csi/README.md:9]: We can make the values configurable in the addon, but if there're too many values to set, the addon will be hard to use. I remember that we're going to support passing values from `Values.yaml` file to the helm type component, we can support passing values from files when enabling the addon in the near future. cc @wangyikewxgm

For the permission thing, I'm not familiar with the azure provider, do they have to install in the `kube-system`?

IMO, I think it's better if we can make users choose whether to install the helm chart or not, but this feature will be supported after kubevela/kubevela#4401 is merged. The trait looks good to me and thank you for your contribution, I think we can merge this PR first and move forward after our feature is ready.

**Review by FogDong — DISMISSED** [review]
Please use `git commit -s` to sign the DCO for your commits. :)

**Comment by Kolossi** [comment]
@FogDong I have rebased to a single commit including DCO signoff, but the check is still not passing.  Please let me know what I should do?

**Review by wonderflow — APPROVED** [review]
thanks

**Pull request merged**
